function_calling/src/tools.py

```diff
@@ -152,40 +152,95 @@ class ReplSchema(BaseModel):
 ## SQL tool
 # tool schema
 class QueryDBSchema(BaseModel):
-    "A database querying tool. Use this to generate sql querys and retrieve the results from a database. Input should be a natural language question to the db."
+    "A query generation tool. Use this to generate sql querys and retrieve the results from a database. Do not pass sql querys directly Input must be a natural language question or instruction."
 
-    query: str = Field(..., description='python code to execute to evaluate')
+    query: str = Field(..., description='natural language question or instruction.')
+
+
+def sql_finder(text):
+    # print(text)
+    sql_code_pattern = re.compile(r'```sql\s+(.*?)\s+```', re.DOTALL)
+    match = sql_code_pattern.search(text)
+    if match:
+        query = match.group(1)
+        return query
+    else:
+        code_pattern = re.compile(r'```\s+(.*?)\s+```', re.DOTALL)
+        match = code_pattern.search(text)
+        if match:
```

Comment:
Same here.

Reply:
fixed

Reply:
match has no len so updated for 

`if match is not None:`